### PR TITLE
How do I submit latency information to the metrics server?

### PR DESCRIPTION
**Pull request by jmorrell**
Given an array of latencies for event loop ticks in milliseconds, how do I submit that to the metrics server?

```
[ 1, 0, 0, 1, 1, 1, 2, 5, 200, ...]
```

There are going to be a lot of these, and most of these are going to be small. @apg suggested in [the quip doc](https://salesforce.quip.com/JLmtAaRYjvVA) to bucket them, so that's what I've done here. This gives me objects that look like:

```
{ '0': 50, '1': 158, '2': 102, '5': 1, '100': 1, '200': 2 }
```

The buckets I've chosen don't matter and we can choose whatever makes sense for the visualization.

cc @freeformz @Xe

### CODE REVIEW

**Comment by apg** [comment]
The easiest thing that would currently work would be to send the metrics as `node.eventloop.latency.$BUCKETNAME`, where `$BUCKETNAME`, perhaps, describes the top of the bucket, e.g. 100, 200, 300, 400, 500, 600. Not sure, but you might want to use log scale for this, so it might actually be 2, 4, 8, 16, 32 ... etc. Statsd (which this is all modeled on) doesn't have an array type, so we can't leverage something like that.

**Comment by jmorrell** [comment]
I've updated the PR so that it's submitting under the metric names: `node.eventloop.latency.%d.ms`

@apg Do you have an idea of what this would look like as a graph? I have something like the latency heatmap for Apollo Optics in my mind, but I'm not sure that's entirely appropriate.

1_0cudtgzkyaleq1rlktdycg

In the end what we want to convey to the user that anything more than a couple of ms is probably a bug, or at least something that should be investigated.

File: src/index.js [hunk 1]
```diff
@@ -99,6 +99,19 @@ setInterval(() => {
   // the metrics data collected above.
   data = {
     counters: {
+      "node.eventloop.latency.0.milliseconds": latencies['0'] || 0,
```
Comment (apg) [inline src/index.js]: Can I suggest `ms` instead of full "milliseconds"?

**Comment by apg** [comment]
@jmorrell That's a good question, and we don't actually have a visualization for it. I have an experimental branch to do a similar thing for request latencies, but no visualization has been built yet, to my knowledge.

The one we've been referring to for inspiration is this one, however:

File: src/index.js (hunk 1 above)
Comment (jmorrell) [reply src/index.js]: Ah, I had that but it didn't make it into the commit

**Comment by jmorrell** [comment]
@apg There is a very different mentality here though. If you have a a few requests that are very slow that's probably fine. However if you have one event loop tick that takes 1 second, that's almost certainly very bad. Whatever visualization we go with should make this clear.

**Comment by jmorrell** [comment]
Updated based on meeting discussion. 

```
{
	"counters": {
		"node.eventloop.latency.0.1.ms": 6,
		"node.eventloop.latency.0.25.ms": 43,
		"node.eventloop.latency.0.5.ms": 67,
		"node.eventloop.latency.1.ms": 137,
		"node.eventloop.latency.2.ms": 744,
		"node.eventloop.latency.4.ms": 376,
		"node.eventloop.latency.8.ms": 1,
		"node.eventloop.latency.16.ms": 0,
		"node.eventloop.latency.32.ms": 0,
		"node.eventloop.latency.64.ms": 0,
		"node.eventloop.latency.128.ms": 0,
		"node.eventloop.latency.256.ms": 0,
		"node.eventloop.latency.512.ms": 0,
		"node.eventloop.latency.1024.ms": 0,
		"node.eventloop.latency.2048.ms": 0,
		"node.eventloop.latency.2048+.ms": 0,
		"node.gc.collections": 4,
		"node.gc.pause.ns": 8632494
	},
	"gauges": {
		"node.eventloop.usage.percent": 0.13952651693749998
	}
}
```

**Pull request merged**